File: cpp/singleton.cpp
#include <map>
#include <iostream>

class DataBase {

    public:

    std::map<const char*, const char*> datas;

    static DataBase* database;

    static DataBase* getInstance() {
        if (!database) DataBase::database = new DataBase();

        return DataBase::database;
    }

    void printValue(const char* key) {
        std::cout << datas[key] << "\n";
    }

    void setValue(const char* key, const char* value) {
        datas[key] = value;
    }

};

DataBase* DataBase::database = 0;

int main() {
    DataBase* data = DataBase::getInstance();

    data->setValue("Лол", "Бомжур!!!");
    data->printValue("Лол");
    DataBase::getInstance()->printValue("Лол");

    return 0;
}

File: py/singleton.py

from typing import Optional


class DataBase:

    _instance = None 

    def __new__(cls) -> "DataBase":
        if not DataBase._instance:
            cls._instance = super().__new__(cls)
            cls._instance.data = {}
        
        return cls._instance # type: ignore
    
    def setValue(self, key: str, value: str):
        self.data[key] = value
    
    def printValue(self, key: str):
        print(self.data[key])

d = DataBase()
d.setValue("Лол", "Бомжур!!!")
d.printValue("Лол")
DataBase().printValue("Лол")
